docs/madr/decisions/056-meshexternalservice-and-zoneegress.md

```diff
@@ -0,0 +1,348 @@
+- it's a bit tricky to set up on universal
+- we've got a bit of a gap in other functionalities on egress (configuration, observability)
+- mTLS is required to run egress
+- there is additional network hop
+- with the introduction of MeshPassthrough the barrier for external traffic is lower and some cases can be covered by that
+
+Conclusion: **it seems like we're not cutting any serious use case**.
+
+#### Running traffic through sidecar only
+
+- all the problems of running through egress are not applicable (operational cost, extra networking hop, scalability etc.)
+- no single exit point for all the external traffic
+- if we put credentials in the ExternalService then these credentials are accessible to the app owners if not delivered via generic_secret
+- it's hard to emulate ExternalService because there is no "from" section (there is no destination that is inside the mesh)
+
+#### Running traffic through sidecar and egress
+
+- we can mix and match the pros/cons of both approaches
+- it's harder to implement / explain how it's working
+
+### Configuration placement
+
+#### Current state and possible options
+
+##### Table summary
+
+The table below shows a summary of policies and if it makes sense to configure them on:
+- sidecar
+- egress
+- on both
+
+and if it makes sense for the user to pick where.
+
+We assume in this table that it never makes sense to put `from` section of a policy on the sidecar because there is no inbound for it, only an outbound (`to` section).
+Both `from` and `to` sections makes sense on the egress.
+
+| Policy                    | On sidecar    | On egress              | On both | User configurable |
+|---------------------------|---------------|------------------------|---------|-------------------|
+| MeshAccessLog             | Yes           | Yes                    | Yes     | Yes               |
+| MeshCircuitBreaker        | Yes           | Maybe                  | No      | No                |
+| MeshFaultInjection        | Maybe         | Yes                    | No      | No                |
+| MeshHealthCheck           | Yes           | Maybe                  | No      | No                |
+| MeshMetric                | Yes           | Yes                    | Yes     | Maybe             |
+| MeshProxyPatch            | Yes           | Yes                    | Yes     | Yes               |
+| MeshRateLimit             | Probably not  | Yes                    | No      | No                |
+| MeshRetry                 | Yes           | Maybe                  | No      | Maybe             |
+| MeshTimeout               | Yes           | Maybe                  | No      | No                |
+| MeshTrace                 | Yes           | Yes                    | Yes     | Maybe             |
+| MeshTrafficPermission     | Maybe         | Yes                    | Maybe   | Maybe             |
+| MeshLoadBalancingStrategy | Yes           | Yes                    | No      | Maybe             |
```

Comment:
ideally this should be on egerss, but we could have validation to only have 'from' if using `proxyType: egress` (similar story with the gateway for fault injection)